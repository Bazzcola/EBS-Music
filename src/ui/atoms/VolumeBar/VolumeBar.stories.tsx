import React from 'react';
import {VolumeBar} from 'ui/atoms/VolumeBar/VolumeBar';
export default {
  title: 'Atoms/VolumeBar',
  component: VolumeBar
};

type active = {
  active:boolean;
}

export const VolumeBar1 = (props:active) => {
    return <VolumeBar active={props.active} />
}