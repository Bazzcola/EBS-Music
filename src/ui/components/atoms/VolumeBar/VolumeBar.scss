input[type=range] {
    width: 100%;
    margin: 4.9px 0;
    background-color: transparent;
    -webkit-appearance: none;
  }
  input[type=range]:focus {
    outline: none;
  }
  input[type=range]::-webkit-slider-runnable-track {
    background: #000000;
    border: 1.1px solid rgba(9, 0, 0, 0);
    border-radius: 6px;
    width: 100%;
    height: 5.2px;
    cursor: pointer;
  }
  input[type=range]::-webkit-slider-thumb {
    margin-top: -4px;
    width: 12px;
    height: 12px;
    background: #fdffff;
    border: 1px solid #808080;
    border-radius: 20px;
    cursor: pointer;
    -webkit-appearance: none;
  }
  input[type=range]:focus::-webkit-slider-runnable-track {
    background: #333333;
  }
  input[type=range]::-moz-range-track {
    background: #000000;
    border: 1.1px solid rgba(9, 0, 0, 0);
    border-radius: 6px;
    width: 100%;
    height: 5.2px;
    cursor: pointer;
  }
  input[type=range]::-moz-range-thumb {
    width: 15px;
    height: 15px;
    background: #fdffff;
    border: 1px solid #808080;
    border-radius: 20px;
    cursor: pointer;
  }
  input[type=range]::-ms-track {
    background: transparent;
    border-color: transparent;
    border-width: 5.9px 0;
    color: transparent;
    width: 100%;
    height: 5.2px;
    cursor: pointer;
  }
  input[type=range]::-ms-fill-lower {
    background: #000000;
    border: 1.1px solid rgba(9, 0, 0, 0);
    border-radius: 12px;
  }
  input[type=range]::-ms-fill-upper {
    background: #000000;
    border: 1.1px solid rgba(9, 0, 0, 0);
    border-radius: 12px;
  }
  input[type=range]::-ms-thumb {
    width: 15px;
    height: 15px;
    background: #fdffff;
    border: 1px solid #808080;
    border-radius: 20px;
    cursor: pointer;
    margin-top: 0px;
  }
  input[type=range]:focus::-ms-fill-lower {
    background: #000000;
  }
  input[type=range]:focus::-ms-fill-upper {
    background: #333333;
  }
  @supports (-ms-ime-align:auto) {
    input[type=range] {
      margin: 0;
    }
  }
  