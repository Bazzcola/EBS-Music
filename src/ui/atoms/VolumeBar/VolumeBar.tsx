import React, { useContext, useRef } from 'react';
import { AudioContext } from 'ui/context/audioContext';
import 'ui/atoms/VolumeBar/VolumeBar.scss';

type active = {
    active:boolean;
}
export const VolumeBar = (props:active) => {
    const { setVolume } = useContext(AudioContext);
    
    const getValue = () => {
        const getInput:any = inputValue.current
        const getClear = getInput.value;
        const getClear2 = Math.floor(getClear);
        setVolume(getClear2 / 10);
    }
    const inputValue = useRef(null);
    console.log(props.active)
    return (
        <input type="range" step="1" min="0" max="10" ref={inputValue} onChange={getValue} defaultValue={5}></input>
   )
}