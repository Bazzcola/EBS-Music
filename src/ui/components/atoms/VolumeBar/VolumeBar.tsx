import React, { useContext, useRef, useCallback } from 'react';
import { AudioContext } from 'ui/context/audioContext';
import 'ui/components/atoms/VolumeBar/VolumeBar.scss';

type active = {
  active?: boolean;
};
export const VolumeBar = (props: active) => {
  const { setVolume } = useContext(AudioContext);

  const getValue = useCallback(() => {
    const getInput: any = inputValue.current;
    const getClear = getInput.value;
    const getClear2 = Math.floor(getClear);
    setVolume(getClear2 / 10);
  }, [setVolume]);
  const inputValue = useRef(null);
  return (
    <div className="input_check">
      <input
        type="range"
        step="1"
        min="0"
        max="10"
        ref={inputValue}
        onChange={getValue}
        defaultValue={5}
      ></input>
    </div>
  );
};
