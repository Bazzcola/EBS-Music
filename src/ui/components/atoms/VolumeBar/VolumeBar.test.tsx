import React from 'react';
import { render } from '@testing-library/react';
import { VolumeBar } from './VolumeBar';

it('renders VolumeBar correctly', () => {
  const { asFragment } = render(<VolumeBar />);
  expect(asFragment()).toMatchSnapshot();
});
