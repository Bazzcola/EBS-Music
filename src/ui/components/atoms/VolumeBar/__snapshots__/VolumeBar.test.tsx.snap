// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders VolumeBar correctly 1`] = `
<DocumentFragment>
  <div
    class="input_check"
  >
    <input
      max="10"
      min="0"
      step="1"
      type="range"
      value="5"
    />
  </div>
</DocumentFragment>
`;
